### PR TITLE
Add tests for `poll_server_for_new_certificates`

### LINKED ISSUE #5541: Tests of `poll_server_for_new_certificates`

### PR DESCRIPTION
**Pull request by TimeEngineer**
closes #5541

### CODE REVIEW

File: libparsec/crates/client/tests/unit/certificates_ops/poll.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+};
+use libparsec_protocol::authenticated_cmds;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use crate::certificates_ops::PollServerError;
+
+use super::utils::certificates_ops_factory;
+
+#[parsec_test(testbed = "minimal", with_server)]
```
Comment (touilleMan) [inline libparsec/crates/client/tests/unit/certificates_ops/poll.rs:14]: Can you avoid using `with_server` ? it's faster and simpler to rely on request mock

File: libparsec/crates/client/tests/unit/certificates_ops/poll.rs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+async fn empty(env: &TestbedEnv) {
+    let alice = env.local_device("alice@dev1");
+
+    let ops = certificates_ops_factory(env, &alice).await;
+
+    test_register_send_hook(
```
Comment (touilleMan) [inline libparsec/crates/client/tests/unit/certificates_ops/poll.rs:20]: If you register a request hook, then you don't need the `with_server` given you won't talk to the server ! ^^

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: libparsec/crates/client/tests/unit/certificates_ops/poll.rs (hunk 2 above)
Comment (TimeEngineer) [reply libparsec/crates/client/tests/unit/certificates_ops/poll.rs:20]: Done

File: libparsec/crates/client/tests/unit/certificates_ops/poll.rs (hunk 1 above)
Comment (TimeEngineer) [reply libparsec/crates/client/tests/unit/certificates_ops/poll.rs:14]: Done

**Review by FirelightFlagboy — APPROVED** [review]

**Review by mmmarcos — APPROVED** [review]
LGTM

File: libparsec/crates/client/tests/unit/certificates_ops/poll.rs [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        Ok(ResponseMock::Mocked((
+            StatusCode::IM_A_TEAPOT,
+            HeaderMap::new(),
+            Bytes::new(),
+        )))
+    });
+
+    ops.poll_server_for_new_certificates(None)
+        .await
+        .unwrap_err();
```
Comment (touilleMan) [inline libparsec/crates/client/tests/unit/certificates_ops/poll.rs:271]: should use `p_assert_matches`

File: libparsec/crates/client/tests/unit/certificates_ops/poll.rs [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+async fn nothing_to_poll(env: &TestbedEnv) {
+    let alice = env.local_device("alice@dev1");
+
+    let ops = certificates_ops_factory(env, &alice).await;
+    let certificates = env
+        .template
+        .certificates()
+        .map(|e| e.signed)
+        .collect::<Vec<_>>();
+    let next_offset = certificates.len() as u64;
```
Comment (touilleMan) [inline libparsec/crates/client/tests/unit/certificates_ops/poll.rs:155]: use `env.get_last_certificate_index()`

**Pull request merged**